packages/unlock-js/src/walletService.ts

```diff
@@ -988,9 +988,10 @@ export default class WalletService extends UnlockService {
       network,
       address: contractAddress,
       abi: passwordHookAbi,
-      signer,
     })
-    const tx = await contract.setSigner(lockAddress, signerAddress)
+    const tx = await contract
+      .connect(this.signer)
```

Comment:
Here since you are sending a transaction (WalletService is to send transactions), then we first need to `connect` the signer to the contract.